src/test/librados/aio_cxx.cc

```diff
@@ -2371,25 +2375,37 @@ TEST(LibRadosAio, PoolEIOFlag) {
 
   bufferlist bl;
   bl.append("some data");
-  std::thread *t = nullptr;
+  std::unique_ptr<std::thread> t;
   std::atomic<bool> missed_eio{false};
   
-  unsigned max = 1000;
-  unsigned timeout = max * 10;
+  auto start_time = std::chrono::steady_clock::now();
+  const int max_run_seconds = 30;
   unsigned long i = 1;
-  my_lock.lock();
-  for (; min_failed == 0 && i <= timeout; ++i) {
+
+  while (true) {
+    {
+      std::unique_lock l(my_lock);
+      if (min_failed != 0) {
+            break;
+      }
+      inflight.insert(i);
+    }
     io_info *info = new io_info;
     info->i = i;
     info->c = Rados::aio_create_completion();
     info->c->set_complete_callback((void*)info, pool_io_callback);
-    inflight.insert(i);
-    my_lock.unlock();
     int r = test_data.m_ioctx.aio_write(test_data.m_oid, info->c, bl, bl.length(), 0);
-    //cout << "start " << i << " r = " << r << std::endl;
+    if (r < 0) {
+      std::cout << "Race caught: aio_write returned " << r << " at index " << i << std::endl;
+      std::scoped_lock l(my_lock);
+      inflight.erase(i);
+      if (inflight.empty()) cv_inflight.notify_all();
+      break;
+    }
 
-    if (i == max / 2) {
-      t = new std::thread([&] {
+    // Trigger EIO after 100 ops have been submitted
+    if (i == 100) {
+      t = std::make_unique<std::thread>([&] {
```

Comment:
Since we already update the code here, how about moving to `std::jthread` to avoid issues like the one here?